src/globus_sdk/transport/retry.py

```diff
@@ -24,23 +25,23 @@ class RetryContext:
     or ``exception`` will be present.
 
     :param attempt: The request attempt number, starting at 0.
+    :param caller_info: Contextual information about the caller, including authorizer
     :param response: The response on a successful request
     :param exception: The error raised when trying to send the request
-    :param authorizer: The authorizer object from the client making the request
     """
 
     def __init__(
         self,
         attempt: int,
         *,
-        authorizer: GlobusAuthorizer | None = None,
+        caller_info: RequestCallerInfo | None = None,
```

Comment:
I don't think the caller info should be nullable. The inner layer (`authorizer`) is, but I don't think there's utility/value in allowing the wrapper itself to be omitted.

`RetryContext` is only constructed inside of the transport layer.

Reply:
This is no longer nullable and all the tests have been updated that broke (~13 tests)